pkg/server/api/agent/v1/service.go

```diff
@@ -83,6 +84,28 @@ func (s *Service) CountAgents(ctx context.Context, req *agentv1.CountAgentsReque
 func (s *Service) ListAgents(ctx context.Context, req *agentv1.ListAgentsRequest) (*agentv1.ListAgentsResponse, error) {
 	log := rpccontext.Logger(ctx)
 
+	parseRequest := func() logrus.Fields {
+		fields := logrus.Fields{}
+
+		if req.Filter != nil {
```

Comment:
This check can be avoided if this function is defined after the `req.Filter != nil` check on line 115